packages/ember-htmlbars/lib/keywords/get.js

```diff
@@ -0,0 +1,108 @@
+import isNone from 'ember-metal/is_none';
+import morphSubscribe from "ember-htmlbars/utils/subscribe";
+import Stream from "ember-metal/streams/stream";
+import SafeString from "htmlbars-util/safe-string";
+import { labelFor } from "ember-metal/streams/utils";
+import { subscribe, read, isStream } from "ember-metal/streams/utils";
+import create from "ember-metal/platform/create";
+import merge from "ember-metal/merge";
+
+function valueStream(obj, key) {
+  var keyValue = read(key);
+  if (isStream(obj) && !isNone(keyValue)) {
+    return obj.get(keyValue);
+  }
+}
+
+function blankValue() {
+  // this is used to compensate for the following line of code
+  //     if (result && result.value) {
+  // in the 'inline' hook. If the stream returns a falsy value after
+  // previously returning a truthy value, this line causes the value not
+  // to be updated
+
+  return new SafeString('');
+}
+
+function getKeywordInline(morph, env, scope, obj, key) {
+  // used when the get keyword is used inline (e.g. {{get obj key}})
+
+  var stream = valueStream(obj, key);
+  morphSubscribe(morph, env, scope, stream);
+
+  var value = env.hooks.getValue(stream);
+
+  if (morph.lastValue !== value) {
+    morph.setContent(value);
+  }
+
+  morph.lastValue = value;
+
+  return true;
+}
+
+function getKeywordSubexpr(obj, key) {
+  // used when the get keyword is used in a subexpression e.g:
+  // {{if (get obj key) 'someValue'}} or {{if someTruth (get obj key)}}
+  // we need to return a Stream object when in a subexpression
+
+  return new GetStream(obj, key);
+}
+
+export default function getKeyword(morph, env, scope, params, hash, template, inverse, visitor) {
+  var objParam = params[0];
+  var keyParam = params[1];
+
+  Ember.assert("The first argument to {{get}} must be a stream", isStream(objParam));
+  Ember.assert("{{get}} requires at least two arguments", params.length > 1);
+
+  if (morph === null) {
+    return getKeywordSubexpr(objParam, keyParam);
+  }
+
+  return getKeywordInline(morph, env, scope, objParam, keyParam);
+}
+
+// function debugLog(getStream, paramName, param) {
+//   var debugLogStreamLabel = Ember.guidFor(getStream)+': '+getStream.label;
+//   console.log(debugLogStreamLabel, paramName, read(param));
+// }
+
+function GetStream(obj, key) {
+  var objLabel = labelFor(obj);
+  var keyLabel = labelFor(key);
+  this.init(`(get ${objLabel} ${keyLabel})`);
+
+  this.sourceDep = this.addMutableDependency(valueStream(obj, key));
+
+  // if (isStream(obj)) {
+  //   subscribe(obj, () => debugLog(this, 'key', obj), this);
+  // }
+
+  if (isStream(key)) {
+    subscribe(key, () => this.updateSource(obj, key), this);
+    // subscribe(key, () => debugLog(this, 'key', key), this);
+  }
```

Comment:
You can compare your implementation with mine (I called it DynamicPathStream instead):

https://gist.github.com/mmun/219ddf4c33ce33e6ad76

The path stream is implicitly subscribed to on line 18. (It's not immediately subscribed to, dependency only set up their subscriptions lazily).

The object stream is indirectly subscribed to via line 39. In a nut shell, the valueDep depends on the object's stream key streams which in turn depend on the object stream itself.